index.js

```diff
@@ -447,24 +447,35 @@ function fastifyMultipart (fastify, options = {}, done) {
   async function saveRequestFiles (options) {
     const requestFiles = []
 
-    const files = await this.files(options)
+    const parts = await this.parts(options)
     this.tmpUploads = []
-    for await (const file of files) {
-      const filepath = path.join(os.tmpdir(), toID() + path.extname(file.filename))
-      const target = createWriteStream(filepath)
-      try {
-        await pump(file.file, target)
-        requestFiles.push({ ...file, filepath })
-        this.tmpUploads.push(filepath)
-        // busboy set truncated to true when the configured file size limit was reached
-        if (file.file.truncated) {
-          const err = new RequestFileTooLargeError()
-          err.part = file
+    if (this.body === undefined) { this.body = {} }
+    for await (const part of parts) {
+      if (part.file) {
+        const filepath = path.join(os.tmpdir(), toID() + path.extname(part.filename))
+        const target = createWriteStream(filepath)
+        try {
+          await pump(part.file, target)
+          requestFiles.push({ ...part, filepath })
```

Comment:
Your result can be archived by a few lines:

```js
if (this.body === undefined) {
    this.body = part.fields
} else {
   this.body = { ...this.body, ...part.fields } 
}
```

Reply:
I agree you can achieve a similar result, but with that you also write back all the files to the body, as they are part of the part.fields attribute.
IMHO this shouldn't be done; if you use saveRequestFiles to write the files to disk they shouldn't also be added to the body. Only the non-file attributes should remain in the body, as they need to be accessible.

Reply:
I agree, then only add non-file attributes.

```js
if (this.body === undefined) {
    this.body = part.fields.map(x => !x.file)
} else {
   this.body = { ...this.body, ...part.fields.map(x => !x.file) } 
}
```

Reply:
Nice use of map. This looks extremely clean. I'll fix it up this way and submit it.

Reply:
@StarpTech Ooof well this was a wild ride. part.fields is an object, not an array.
Honestly I'd recommend sticking with my original code. Everything else is just shitshow of case handling.

Reply:
What's wrong with that?
```js
for (var fieldName in part.fields) {
  const field = this.body[fieldName]
  if (this.body[fieldName].hasOwnProperty(fieldName) && !field.file) {
	this.body[fieldName] = field
  }
}
```

Reply:
What irks me about this is the fact the we are already iterating over all the parts and accessing the .fields property. This only needs to happen once, but this way it happens on every file in the request.
Imagine a scenario where 10 files are in a request, then this is executed ten times where one would have sufficed...

Reply:
I agree, I'm also fine with.

```js
this.body = { ...this.body, ...part.fields }
```

Reply:
Then just do:

```js
this.body = { ...this.body, ...part.fields } 
```

Reply:
So then we are back at square zero where we have a duplication of information: On the one hand, the files from the multipart form have been written to disk, on the other hand they are accessible through the body again.

What exactly is so bad about using the parts iterator and my approach in adding the fields to the body? To me it seems like the only logical thing given the desired result of having only the non-file fields in the body and doing so without a performance penalty for requests with a large number of parts.

Reply:
It looks odd because we already collected all fields before in the same way. https://github.com/fastify/fastify-multipart/blob/ccf96b0cd138593fd4b1ce1c0bcf2a46ac816cb2/index.js#L423-L429